client/components/token-field/token-input.jsx

```diff
@@ -17,21 +18,24 @@ var TokenInput = React.createClass( {
 			onChange: function() {},
 			onBlur: function() {},
 			value: '',
-			disabled: false
+			disabled: false,
+			placeholder: '',
 		};
 	},
 
 	mixins: [ PureRenderMixin ],
 
 	render: function() {
 		const props = { ...this.props, onChange: this._onChange };
+		const { value, placeholder } = props;
+		const size = ( placeholder && placeholder.length || value.length ) + 1;
```

Comment:
I find I can never remember the order of mixed operators like this. Would be good to include some parentheses around the `&&` expression.

Separately I think it'd be nice to enable the [`no-mixed-operators`](http://eslint.org/docs/rules/no-mixed-operators) ESLint rule.